src/python/pants/engine/target.py

```diff
@@ -59,8 +61,7 @@
 
 
 @frozen_after_init
-@dataclass(unsafe_hash=True)
-class Field(ABC):
+class Field:
```

Comment:
I don't think subclassing `ABC` pulls its weight. We weren't defining `@abstractmethod` anywhere, so this didn't even do anything.

There's an argument that it expresses the intent that you should subclass it, but I don't think that's worth the code cost. We can simply document that you should subclass.

Reply:
The `alias` and `default` could be `@abstractclassproperty`? I guess right now we're relying on mypy to assert that they are set.

Reply:
I'd rather avoid it - `@classproperty` doesn't work nicely with this style, it expects you to do the `@classproperty` method:

```python
class MyField(Field):
  alias = "field"
```